File: backend/src/main/java/com/attendme/dto/requestdto/AttendanceRequest.java
package com.attendme.dto.requestdto;

import jakarta.validation.constraints.NotNull;
import lombok.Data;

import java.time.LocalDate;

@Data
public class AttendanceRequest {
    
    @NotNull(message = "Class ID is required")
    private Long classId;
    
    @NotNull(message = "Student ID is required")
    private Long studentId;
    
    @NotNull(message = "Date is required")
    private LocalDate date;
    
    @NotNull(message = "Status is required")
    private String status;
    
    private String remarks;
    
    @NotNull(message = "Marked by user ID is required")
    private Long markedById;
}

File: mobile/AttendMe/app/src/main/java/edu/cit/cararag/attendme/data/model/AttendanceRequest.kt
package edu.cit.cararag.attendme.data.model

data class AttendanceRequest(
    val studentId: Long,
    val classId: Long,
    val date: String,       // format: yyyy-MM-dd
    val status: String,     // PRESENT, ABSENT, LATE, EXCUSED
    val remarks: String? = null
)